backfill/2020-01-23-sync-ping/README.md

```diff
@@ -0,0 +1,65 @@
+# Backfilling sync pings due to unexpected devices fields
+
+See https://bugzilla.mozilla.org/show_bug.cgi?id=1610907
+
+## Steps
+
+We ran the backfill in the `moz-fx-data-backfill-4` project. 
+
+First, we determine the backfill range by querying the relevant error table:
+
+```
+SELECT
+  date(submission_timestamp) as dt, count(*)
+FROM
+  `moz-fx-data-shared-prod.payload_bytes_error.telemetry`
+WHERE
+  DATE(submission_timestamp) >= "2020-01-10"
+  AND document_type = 'sync'
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/payload/devices/%: extraneous key % is not permitted'
+  group by 1 order by 1
+```
+
+That showed the affected range as `2020-01-20` through `2020-01-23`.
+
+Next, we create destination tables via the `mirror-prod-tables` script and
+we populate the `backfill_input` table via query:
+
+```
+SELECT
+  *
+FROM
+  `moz-fx-data-shared-prod.payload_bytes_error.telemetry`
+WHERE
+  DATE(submission_timestamp) BETWEEN "2020-01-20" AND "2020-01-23"
+  AND document_type = 'sync'
+  AND error_message LIKE 'org.everit.json.schema.ValidationException: #/payload/devices/%: extraneous key % is not permitted'
+```
+
+Next, we construct a suitable Dataflow job configuration in
+`launch-dataflow-sync-ping` and run the script.
+
+We visit the GCP console, choose the `moz-fx-data-backfill-4` project
+and go to the Dataflow section to watch the progress of the job.
+It took about X minutes to run to completion.
+
+We validate the results by checking counts per day and also by checking whether
+we have any overlapping IDs between prod and the backfilled table:
```

Comment:
> checking whether we have any overlapping IDs between prod and the backfilled table

Is this check missing?

Reply:
We actually don't need to worry about that part for this scenario since we're backfilling to live rather than stable. This can be removed